docs/source/matchers_tutorial.ipynb

```diff
@@ -0,0 +1,377 @@
+   },
+   "source": [
+    "Lets try to do a bit better with matchers. We can make a better function that takes advantage of matchers to get rid of both the instance check and the `ensure_type` call, like so:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import libcst.matchers as m\n",
+    "\n",
+    "def better_is_call_with_booleans(node: cst.Call) -> bool:\n",
+    "    for arg in node.args:\n",
+    "        if not m.matches(arg.value, m.Name(\"True\") | m.Name(\"False\")):\n",
+    "            # Oops, this isn't a True/False literal!\n",
+    "            return False\n",
+    "    # We got here, so all arguments are literal boolean values.\n",
+    "    return True\n"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "This is a lot shorter and is easier to read as well! We made use of the fact that matchers handles instance checking for us in a safe way. We also made use of the fact that matchers allows us to concisely express multiple match options with the use of Python's or operator. We can also see that it still works on our previous examples:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "better_is_call_with_booleans(call_1)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "better_is_call_with_booleans(call_2)\n"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "We still have one more trick up our sleeve though. Matchers don't just allow us to specify which attributes we want to match on exactly. It also allows us to specify rules for matching sequences of nodes, like the list of ``Arg`` nodes that appears in ``Call``. Lets make use of that, turning our original ``is_call_with_booleans`` function into a call to ``matches``:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def best_is_call_with_booleans(node: cst.Call) -> bool:\n",
+    "    return m.matches(\n",
+    "        node,\n",
+    "        m.Call(\n",
+    "            args=(\n",
+    "                m.ZeroOrMore(m.Arg(m.Name(\"True\") | m.Name(\"False\"))),\n",
+    "            ),\n",
+    "        ),\n",
+    "    )\n"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "We've turned our original function into a single call to ``matches``. As an added benefit, the match node can be read from left to right in a way that makes sense in english: \"match any call with zero or more arguments that are the literal ``True`` or ``False``.\". As we can see, it works as intended: "
```

Comment:
Do we need the dot inside the double quotes? It looks odd in the generated doc. 
![image](https://user-images.githubusercontent.com/3840867/65985568-ee5dc380-e436-11e9-929f-c6adc39ebcff.png)

Reply:
I think that's a copy mistake, I can fix it.